### ISSUE TITLE
Protocol of FRCNN retraining

### ISSUE THREAD

**Issue by vadimkantorov** [ppengtang/pcl.pytorch#66]
Hi! What is the protocol of "+FRCNN" retraining mentioned in the paper? How do you generate pseudo ground truth for this retraining?

Does this repo contain scripts for this retraining? 

I'm interested to try different detectors / architectures  to do this retraining to check how they fare against FRCNN.

Thank you!

**Comment by hhhh12345678** [comment]
Hi! Coincidentally, I'm also thinking about the protocl of '+FRCNN'.

Based on the codes provided, i got 52.1mAP. But when i used the VOC2007trainval pseudo ground truth to retrain FRCNN, 
I only got 45.1mAP. it is seems very strange, since the trick is used as a means to increase performance in many papers.

Have you try the retraining step?

**Comment by bityangke** [comment]
if u use oicr+reg config, i think no need to retrain FRCNN

**Comment by ppengtang** [comment]
Hi @vadimkantorov and @hhhh12345678 , I haven't tried "+FRCNN" based on this new version of codes. I only tried "+FRCNN" for the caffe version of pcl/oicr. I simply chose the box w/ max score for each positive class as the pseudo gt for FRCNN training (https://github.com/ppengtang/oicr/blob/master/lib/fast_rcnn/test_train.py#L245-L251).

Some tips to train better FRCNN:
1. Multi-scale training and testing for FRCNN.
2. Dilation for the 5th conv stage.
3. Sampling large number of proposals per-image during training (e.g., 2000).
4. Fast R-CNN instead of Faster R-CNN (not sure whether this is still true with better WSOD results).
See https://github.com/ppengtang/fast-rcnn/blob/master/lib/fast_rcnn/config.py for some configurations.

Hope these could be helpful.
